backend/tarsy/agents/base_agent.py

```
@@ -382,50 +424,83 @@ async def determine_next_mcp_tools(self,
+        
         try:
+            # Extract data using type-safe helper methods
+            runbook_content = alert_data.get_runbook_content()
+            original_alert = alert_data.get_original_alert_data()
+            
+            # Get accumulated MCP data from all previous stages
+            initial_mcp_data = alert_data.get_all_mcp_results()
+            stage_attributed_mcp_data = alert_data.get_stage_attributed_mcp_results()
+            
+            # Log enriched data usage from previous stages
+            if previous_data := alert_data.get_stage_result("data-collection"):
```

Ruff (0.12.2)

Local variable `previous_data` is assigned to but never used

(F841)